bigquery/scheduled_queries/calculate-daily-job-alert-metrics.sql

```diff
@@ -9,21 +9,52 @@ WITH
     SELECT
       date
     FROM
-      `teacher-vacancy-service.production_dataset.CALCULATED_job_alert_metrics`)),
+      `teacher-vacancy-service.production_dataset.CALCULATED_job_alert_metrics`)
+    OR date >= CURRENT_DATE - 7),
+  #recalculate the last 7 days' worth of data each night so that we pick up job alert email CTRs for up to a week after sending the email, but then fix this data
   job_alert_emails_sent AS (
   SELECT
-    run_on AS date,
-    COUNT(*) AS number_of_alert_emails_sent
+    alertrun.run_on AS date,
+    COUNT(DISTINCT alertrun.id) AS number_of_alert_emails_sent,
+    COUNT(DISTINCT
+    IF
+      ("vacancy" IN UNNEST(users_from_job_alert_emails.job_alert_destinations),
+        users_from_job_alert_emails.alertrun_id,
+        NULL)) AS number_of_alert_emails_clicked_on,
+    COUNT(DISTINCT
+    IF
+      ("unsubscribe" IN UNNEST(users_from_job_alert_emails.job_alert_destinations),
+        users_from_job_alert_emails.alertrun_id,
+        NULL)) AS number_of_alert_emails_unsubscribed_from,
+    COUNT(DISTINCT
+    IF
+      ("edit" IN UNNEST(users_from_job_alert_emails.job_alert_destinations),
+        users_from_job_alert_emails.alertrun_id,
+        NULL)) AS number_of_alert_emails_edited
   FROM
     `teacher-vacancy-service.production_dataset.feb20_alertrun` AS alertrun
+  LEFT JOIN (
+    SELECT
+      utm_source AS alertrun_id,
+      job_alert_destinations
+    FROM
+      `teacher-vacancy-service.production_dataset.CALCULATED_daily_users_from_cloudfront_logs`
+    WHERE
+      from_job_alert
+      AND utm_source IS NOT NULL
+      AND utm_source != "subscription") AS users_from_job_alert_emails
```

Comment:
This is because we used to have the utm_source set to the string 'subscription', correct?

Reply:
Yes - and we need to distinguish between past versions of the data and current versions.